### PR TITLE
Part of #11437: moved window_setup_file_association(s) in Platform2.hpp

### PR DESCRIPTION
**Pull request by frutiemax**
window_setup_file_association(s) now become SetupFileAssociation(s). I'd like to get a good review on this one as I'm not too sure if I did it correctly.

### CODE REVIEW

File: src/openrct2/platform/Platform.Win32.cpp [hunk 1]
```diff
@@ -347,6 +351,129 @@ namespace Platform
...
+        {
+            RegCloseKey(hKey);
+            RegCloseKey(hRootKey);
+            return false;
+        }
+
+        // [hRootKey\.ext]
+        if (RegSetValueW(hRootKey, extensionW.c_str(), REG_SZ, progIdNameW.c_str(), 0) != ERROR_SUCCESS)
+        {
+            RegCloseKey(hKey);
```
Comment (duncanspumpkin) [inline src/openrct2/platform/Platform.Win32.cpp:418]: hKey isn't opened at this point.

File: src/openrct2/platform/Platform.Win32.cpp [hunk 2]
```diff
@@ -347,6 +351,129 @@ namespace Platform
...
+        auto commandArgsW = String::ToWideChar(commandArgs);
+        auto progIdNameW = get_progIdName(extension);
+
+        HKEY hKey = nullptr;
+        HKEY hRootKey = nullptr;
+
+        // [HKEY_CURRENT_USER\Software\Classes]
+        if (RegOpenKeyW(HKEY_CURRENT_USER, SOFTWARE_CLASSES, &hRootKey) != ERROR_SUCCESS)
+        {
+            RegCloseKey(hKey);
```
Comment (duncanspumpkin) [inline src/openrct2/platform/Platform.Win32.cpp:410]: hKey isn't opened at this point.

File: src/openrct2/platform/Platform.Win32.cpp [hunk 3]
```diff
@@ -347,6 +351,129 @@ namespace Platform
...
+    bool SetUpFileAssociation(
+        const std::string extension, const std::string fileTypeText, const std::string commandText,
+        const std::string commandArgs, const uint32_t iconIndex)
+    {
+        wchar_t exePathW[MAX_PATH];
+        wchar_t dllPathW[MAX_PATH];
+
+        [[maybe_unused]] int32_t printResult;
+
+        GetModuleFileNameW(nullptr, exePathW, (DWORD)std::size(exePathW));
```
Comment (duncanspumpkin) [inline src/openrct2/platform/Platform.Win32.cpp:395]: Use static casts instead of c style casts

File: src/openrct2/platform/Platform.Win32.cpp [hunk 4]
```diff
@@ -35,6 +37,8 @@
 
 #    include <iterator>
 
+#    define SOFTWARE_CLASSES L"Software\\Classes"
```
Comment (duncanspumpkin) [inline src/openrct2/platform/Platform.Win32.cpp:40]: Use a constexpr instead of a macro

File: src/openrct2/platform/Platform.Win32.cpp [hunk 5]
```diff
@@ -347,6 +351,129 @@ namespace Platform
...
+        auto fileTypeTextW = String::ToWideChar(fileTypeText);
+        auto commandTextW = String::ToWideChar(commandText);
+        auto commandArgsW = String::ToWideChar(commandArgs);
+        auto progIdNameW = get_progIdName(extension);
+
+        HKEY hKey = nullptr;
+        HKEY hRootKey = nullptr;
+
+        // [HKEY_CURRENT_USER\Software\Classes]
+        if (RegOpenKeyW(HKEY_CURRENT_USER, SOFTWARE_CLASSES, &hRootKey) != ERROR_SUCCESS)
```
Comment (duncanspumpkin) [inline src/openrct2/platform/Platform.Win32.cpp:408]: Technically these functions shouldn't be used all the reg functions should be using the exw versions. But that's for a future refactor.

**Comment by Gymnasiast** [comment]
Quite happy you got rid of a few gotos as well :+1:

**Pull request merged**